docs/docs/testing.md

```diff
@@ -1744,6 +1744,225 @@ Luckily, RedwoodJS has several api testing utilities to make [testing functions
+import '@redwoodjs/testing/cache'
+// ^^ make `.toHaveCached` available
+```
+
+The `toHaveCached` matcher can take three forms:
+
+`expect(testCacheClient)`
+1. `.toHaveCached(expectedData)` - check for an exact match of the data, regardless of the key
+2. `.toHaveCached('expected-key', expectedData)` - check that the data is cached in the key you supply
+3. `.toHaveCached(/key-regex.*/, expectedData)` - check that data is cached in a key that matches the regex supplied
+
+
+Let's see these in action now:
+
+```ts
+
+scenario('returns a single product', async (scenario: StandardScenario) => {
+  await product({ id: scenario.product.three.id })
+
+// Pattern 1: Only check that the data is present in the cache
+  expect(testCacheClient).toHaveCached(scenario.product.three)
+
+// Pattern 2: Check that data is cached, at a specific key
+   expect(testCacheClient).toHaveCached(
+      `cached-product-${scenario.product.three.id}`,
+      scenario.product.three
+    )
+
+// Pattern 3: Check that data is cached, in a key matching the regex
+  expect(testCacheClient).toHaveCached(
+      /cached-.*/,
+      scenario.product.three
+    )
+```
+
+
+:::info Serialized Objects in Cache
+Remember that the cache only ever contains serialized objects. So if you passed an object like this:
+```js
+{
+  id: 5,
+  published: new Date('12/10/1995')
+}
+
+```
+
+The published key will be serialized and stored as a string. To make testing easier for you, we serialize the object you are passing when you use the `toHaveCached` matcher, before we compare it against the value in the cache
+:::
+
+### Partial Matching
+It can be a little tedious to check that every key in the object you are looking for matches. This is especially true if you have autogenerated values such as `updatedAt` and `cuid` Ids.
+
+To help with this, we've provided a helper for partial matching!
+
+```ts
+// highlight-next-line
+import { partialMatch } from '@redwoodjs/testing/cache'
+
+
+scenario('returns all products', async (scenario: StandardScenario) => {
+  await products()
+
+ // Partial match using the toHaveCached, if you supply a key
+   expect(testCacheClient).toHaveCached(
+    /cached-products.*/,
+    // highlight-next-line
+    partialMatch([{ name: 'LS50', brand: 'KEF' }])
+  )
+
+  // Or you can use the .contents getter
+  expect(testCacheClient.contents).toContainEqual(
+  // check that an array contains an object matching
+  // highlight-next-line
+   partialMatch([{ name: 'LS50', brand: 'KEF' }])
+  )
+}
+
+scenario('finds a single product', () = {
+   await product({id: 5})
+
+  // You can also check for a partial match of an object
+   expect(testCacheClient).toHaveCached(
+      /cached-.*/,
+  // highlight-start
+      partialMatch({
+        name: 'LS50',
+        brand: 'KEF'
+      })
+    )
+  // highlight-end
+})
+
+```
+
+Partial match is just syntactic sugar - underneath it just uses Jest's `expect.objectContaining` and `expect.arrayContaining`.
```

Comment:
```suggestion
Partial match is just syntactic sugar—underneath it uses Jest's `expect.objectContaining` and `expect.arrayContaining`.
```

Reply:
I'll acept but I'm not sure without spaces is the correct one ;)